spinalcordtoolbox/scripts/sct_create_mask.py

```diff
@@ -242,23 +242,13 @@ def create_mask(param):
     for iz in range(0, nz, 1):
         if iz in z_centerline_not_null:
             cx[iz], cy[iz] = ndimage.measurements.center_of_mass(np.array(data_centerline[:, :, iz]))
-    # create 2d masks
-    file_mask = 'data_mask'
+
+    # create 2d masks at each axial slice
+    im_out = zeros_like(im_data)
     for iz in range(nz):
-        if iz not in z_centerline_not_null:
-            # write an empty nifty volume
-            img = nibabel.Nifti1Image(data_centerline[:, :, iz], None, hdr)
-            nibabel.save(img, (file_mask + str(iz) + '.nii'))
-        else:
+        if iz in z_centerline_not_null:
             center = np.array([cx[iz], cy[iz]])
-            mask2d = create_mask2d(param, center, param.shape, param.size, im_data=im_data)
-            # Write NIFTI volumes
-            img = nibabel.Nifti1Image(mask2d, None, hdr)
-            nibabel.save(img, (file_mask + str(iz) + '.nii'))
```

Comment:
In the old code, when we were saving and reloading these `Nifti1Images`, under the hood nibabel was applying all kinds of logic to the headers. (For example, it was replacing `None` with `header.get_best_affine()` when loading the image.)

So, unfortunately it's not equivalent to do `im_out = zeros_like(im_data)` and `im_out.data[:, :, iz] = create_mask_2d()`.